File: cpp/fibonacci.cpp
/*
 * fibonacci.cpp
 */


#include <iostream>

using namespace std;

int fib_iter(int n)
{
    if ( n == 0)
        return 0;
    else if ( n == 1)
        return 1;
    int a = 0;
    int b = 1;
    int tmp;
    cout << a << endl;
    for (int i = 1 ; i < n; i++) 
    {   tmp = b;
        b = a + b;
        a = tmp;
        cout << a << " " << b << " " << "Iloraz: " << b / a << endl;
    }
    cout << a << " " << b << " " << "Iloraz: " << b / a << endl;
    return b;
}

int fib_rek(int n)
{
    if (n < 2)
    {
        return 1;
    }
    return fib_rek(n - 2) + fib_rek(n - 1);
}

int main(int argc, char **argv)
{
    int n = 0;
    cout << "Numer wyrazu ciągu: " << endl;
    cin >> n;
    cout << fib_iter(n) << endl; 
    cout << "  " << endl; 
    cout << fib_rek(n) << endl; 
       
    return 0;
}

File: python/fibonacci.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


def fib_rek(n):
    if n < 2:
        return 1
    return fib_rek(n - 2) + fib_rek(n - 1)


def fib_iter(n):
    """Funkcja wyświetla kolejne wyrazy ciągu Fibonacciego.
    Funkcja zwraca n-ty wyraz ciągu.
    F(0) = 0
    F(1) = 1
    F(n) = F(n-2) + F(n-1) dla n > 1
    """
    a, b = (0, 1)
    if n == 0:
        print (a)
        return a
    # elif n == 1:
        # print (b)
        # return b
    print(a)
    for i in range(2, n + 2):
        # tmp = b
        # b = a + b
        # a = tmp
        a, b = b, a + b
        print(a, "Wyraz ", i, ": ", b, "Iloraz:", b / a)
    return b


def fib_iter2(n):
    a, b = (0, 1)

    print (a)
    while n > 0:
        a, b = b, a + b
        print(a, "", b, "Iloraz: ", b / a)
        n = n - 1
    return b


def main(args):
    n = int(input("Podaj ilość liczb w ciągu"))

    print (fib_rek(n))
    # print (fib_iter(n))
    # print (fib_iter2(n))
    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
